community/command-line/src/main/java/org/neo4j/commandline/admin/Usage.java

```diff
@@ -40,75 +39,53 @@ public Usage( String scriptName, CommandLocator commands )
-        providers.forEach( command ->
-        {
-            final CommandUsage commandUsage = new CommandUsage( command, scriptName );
-            commandUsage.printIndentedSummary( output );
-        } );
-    }
-
-    public void printUsageForCommand( AdminCommand.Provider command, Consumer<String> output )
-    {
-        final CommandUsage commandUsage = new CommandUsage( command, scriptName );
-        commandUsage.printDetailed( output );
+        output.accept( "    NEO4J_CONF    Path to directory which contains neo4j.conf." );
+        output.accept( "    NEO4J_DEBUG   Set to anything to enable debug output." );
+        output.accept( "    NEO4J_HOME    Neo4j home directory." );
+        output.accept( "" );
     }
 
-    public static class CommandUsage
+    private void printCommands( Consumer<String> output )
     {
-        private final AdminCommand.Provider command;
-        private final String scriptName;
-
-        public CommandUsage( AdminCommand.Provider command, String scriptName )
-        {
-            this.command = command;
-            this.scriptName = scriptName;
-        }
-
-        public void printSummary( Consumer<String> output )
-        {
-            output.accept( format( "%s", command.name() ) );
-            output.accept( "    " + command.summary() );
-        }
-
-        public void printIndentedSummary( Consumer<String> output )
-        {
-            printSummary( s -> output.accept( "    " + s ) );
-        }
+        Map<AdminCommandSection,List<AdminCommand.Provider>> groupedProviders = groupProvidersBySegment();
 
-        public void printDetailed( Consumer<String> output )
-        {
-            for ( Arguments arguments : command.possibleArguments() )
-            {
-                //Arguments arguments = command.arguments();
+        AdminCommandSection.general()
+                .printAllCommandsUnderSection( output, groupedProviders.remove( AdminCommandSection.general() ) );
 
-                String left = format( "usage: %s %s", scriptName, command.name() );
+        groupedProviders.entrySet().stream()
+                .sorted( Comparator.comparing( groupedProvider -> groupedProvider.getKey().printable() ) )
+                .forEach(entry -> entry.getKey().printAllCommandsUnderSection( output, entry.getValue() ) );
+    }
 
-                output.accept( Arguments.rightColumnFormatted( left, arguments.usage(), left.length() + 1 ) );
-            }
-            output.accept( "" );
-            output.accept( command.allArguments().description( command.description() ) );
-        }
+    private Map<AdminCommandSection,List<AdminCommand.Provider>> groupProvidersBySegment()
+    {
+        List<AdminCommand.Provider> providers = new ArrayList<>();
+        commands.getAllProviders().forEach( providers::add );
+        return providers.stream().collect( Collectors.groupingBy( AdminCommand.Provider::commandSection ) );
```

Comment:
> Now it becomes impossible for any other command to be listed under the capi flash section unless it shares source with the other capiflash commands.

Shouldn't this be expected of them anyway if you are developing stuff related to capi flash it should depend on capi flash module thereby inherit the CapiFlashCommandSection that comes along with it.